internal/cmd/use.go

```diff
@@ -60,8 +78,28 @@ func validateVersionArg(c *cli.Context) error {
 		return NewValidationError("missing [version] argument")
 	}
 	v := c.Args().First()
-	if matched := globals.EnvoyVersionPattern.MatchString(v); !matched {
+	if matched := globals.EnvoyMinorVersionPattern.MatchString(v); !matched {
 		return NewValidationError("invalid [version] argument: %q should look like %q", v, version.LastKnownEnvoy)
 	}
 	return nil
 }
+
+func getLatestInstalledPatch(o *globals.GlobalOpts, minorVersion version.Version) (version.Version, error) {
+	rows, err := getInstalledVersions(o.HomeDir)
+	if err != nil {
+		return "", err
+	}
+	// Sort so that new release dates appear first and on conflict choosing the higher version.
+	sort.Slice(rows, func(i, j int) bool {
+		if rows[i].releaseDate == rows[j].releaseDate {
+			return rows[i].version > rows[j].version
+		}
+		return rows[i].releaseDate > rows[j].releaseDate
+	})
+	for i := range rows {
+		if strings.HasPrefix(string(rows[i].version), string(minorVersion)+".") {
```

Comment:
May we could avoid allocating the string in every loop 😄 
```suggestion
	prefix := string(minorVersion)+"."
	for i := range rows {
		if strings.HasPrefix(string(rows[i].version), prefix) {
```

Reply:
```suggestion
	for _, row := range rows {
		if strings.HasPrefix(string(row.version), string(minorVersion)+".") {
```

Reply:
Ah OK